plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,217 @@
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+
+        this.partitionColumns = getPartitionColumns(new SchemaTableName(tableName.getSchemaName(), DeltaLakeTableName.tableNameFrom(tableName.getTableName())), this.tableLocation, session);
+
+        List<ColumnMetadata> columns = partitionColumns.stream()
+                .map(column -> new ColumnMetadata(column.getBaseColumnName(), column.getType()))
+                .collect(toImmutableList());
+
+        this.tableMetadata = new ConnectorTableMetadata(tableName, columns);
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session,
+                                          TupleDomain<Integer> constraint)
+    {
+        TableSnapshot tableSnapshot;
+        MetadataEntry metadataEntry;
+        try {
+            // Verify the transaction log is readable
+            SchemaTableName baseTableName = new SchemaTableName(tableName.getSchemaName(),
+                    DeltaLakeTableName.tableNameFrom(tableName.getTableName()));
+            tableSnapshot = transactionLogAccess.loadSnapshot(session, baseTableName, tableLocation);
+            metadataEntry = transactionLogAccess.getMetadataEntry(tableSnapshot, session);
+        }
+        catch (IOException e) {
+            throw new TrinoException(DELTA_LAKE_INVALID_SCHEMA,
+                    "Unable to load table metadata from location: " + tableLocation, e);
+        }
+        if (partitionColumns.isEmpty()) {
+            return new EmptyPageSource();
+        }
+        return new FixedPageSource(buildPages(session, tableSnapshot, metadataEntry));
+    }
+
+    private List<Page> buildPages(ConnectorSession session, TableSnapshot tableSnapshot, MetadataEntry metadataEntry)
+    {
+        PageListBuilder pageListBuilder = PageListBuilder.forTable(tableMetadata);
+
+        Stream<AddFileEntry> activeFiles = transactionLogAccess.loadActiveFiles(tableSnapshot, metadataEntry,
+                transactionLogAccess.getProtocolEntry(session, tableSnapshot), TupleDomain.all(), alwaysTrue(), session);
+
+        for (Map<String, Optional<String>> partitionValue : getDedupedPartitionValues(activeFiles)) {
+            if (partitionValue.isEmpty()) {
+                pageListBuilder.beginRow();
+                pageListBuilder.appendNull();
+                pageListBuilder.endRow();
+                continue;
+            }
+
+            pageListBuilder.beginRow();
+            for (DeltaLakeColumnHandle column : partitionColumns) {
+                String actualPartitionValue = partitionValue.get(column.getBaseColumnName()).orElse(null);
```

Comment:
we should probably use the actual partition value type see `io.trino.plugin.deltalake.transactionlog.TransactionLogParser#deserializePartitionValue`